utils/utils.py

```diff
@@ -0,0 +1,16 @@
+import re
+
+bad_chars = re.compile(r"[^A-Za-z0-9_. ]+|^\.|\.$|^ | $|^$")
+bad_names = re.compile(r"(aux|com[1-9]|con|lpt[1-9]|prn)(\.|$)")
+
+
+def sanitize_name(s):
```

Comment:
it doesn't so much of sanitization -- it is kind of creating a filename. I suggest removing this function instead, using the book id to write it to a file.